docs(kic): add compatibilities of router flavors of Kong (#4834)
Kong/docs.konghq.com: src/kubernetes-ingress-controller/references/supported-router-flavors.md

---
title: Supported Kong Router Flavors
---

> ## Supported Kong Router Flavors

Reviewer: ```suggestion
```
This title is just a duplicate of the page title, so it's not needed.

> {{site.base_gateway}} (open-source and enterprise edition) has included the new [expression based router][gateway-expression-router] in version 3.0 and later. 

Reviewer: ```suggestion
{{site.base_gateway}} (open-source and enterprise edition) includes an [expression-based router][gateway-expression-router] in versions 3.0 and later. 
```

Author: fixed

> The router of {{site.base_gateway}} could be configured to the following [modes][gateway-router-flavor]:

Reviewer: ```suggestion
The router can be configured in the following [modes][gateway-router-flavor]:
```

- `traditional`: uses the pre-3.0 router.

> - `traditional_compatible`: uses the expression based router, but the rotuer configure interface remains the same as `traditional`.

Reviewer: ```suggestion
- `traditional_compatible`: uses the expression based router, but the router configuration interface remains the same as `traditional`.
```

Author: fixed

> - `expression`: uses expression based rputer, and expressions must be provided in router configure interface.

Reviewer: ```suggestion
- `expression`: uses the expression-based router, and expressions must be provided in the router configuration interface.
```

Author: fixed

> The compatabilities of router flavors between different {{site.kic_product_name}} version and {{site.base_gateway}} is shown in the following table.

Reviewer: ```suggestion
The compatibilities of router flavors between different {{site.kic_product_name}} versions and {{site.base_gateway} are shown in the following table.
```

Author: fixed

| {{site.kic_product_name}}           | 2.7.x                           |
|:-----------------------------------:|:-------------------------------:|

> | Kong 3.0.x  traditional             |  <i class="fa fa-check"></i>    |     

Reviewer: ```suggestion
| Kong 3.0.x  `traditional`            |  <i class="fa fa-check"></i>    |     
```

Author: fixed

> | Kong 3.0.x  traditional_compatible  |  <i class="fa fa-check"></i>(*) | 

Reviewer: ```suggestion
| Kong 3.0.x  `traditional_compatible`  |  <i class="fa fa-check"></i>(*) | 
```

> | Kong 3.0.x  expression              |  <i class="fa fa-times"></i>    |

Reviewer: ```suggestion
| Kong 3.0.x  `expression`              |  <i class="fa fa-times"></i>    |
```

Author: fixed

> (*) Most use cases are supported. Regexed with a backslash (`\`) followed by a non-escaped character (for example `\j`) in matches of paths or headers

Reviewer: ```suggestion
(*) Most use cases are supported. Regexes with a backslash (`\`) followed by a non-escaped character (for example, `\j`) in matches of paths or headers
```

Author: fixed

> may not be accepted by Kong 3.0 configured to use `traditional_compatible` router.

Reviewer: ```suggestion
may not be accepted when {{site.base_gateway}} 3.0 is configured to use the `traditional_compatible` router.
```

Author: fixed
